tests/functional/aws-node-sdk/lib/utility/website-util.js

```diff
@@ -36,91 +236,61 @@ export class WebsiteConfigTester {
-                    browser.assert.text('ul:first-of-type > li:nth-child(2)',
-                    'Message: Access Denied');
-                    browser.assert.text('h3', 'An Error Occurred While ' +
-                    'Attempting to Retrieve a Custom Error Document');
-                    browser.assert.text('ul:nth-of-type(2) > li:first-child',
-                    'Code: AccessDenied');
-                    browser.assert.text('ul:nth-of-type(2) > li:nth-child(2)',
-                    'Message: Access Denied');
-                } else {
-                    throw new Error('This 403 error is not checked in ' +
-                    'checkHTML()');
-                }
-            } else if (type === 'index-user') {
-                browser.assert.status(200);
-                browser.assert.text('title',
-                    'Best testing website ever');
-                browser.assert.text('h1', 'Welcome to my ' +
-                    'extraordinary bucket website testing page');
-            } else if (type === '200') {
-                browser.assert.status(200);
-            } else if (type === 'error-user') {
-                browser.assert.status(403);
-                browser.assert.text('title', 'Error!!');
-                browser.assert.text('h1', 'It appears you messed up');
-            } else if (type === 'error-user-404') {
-                browser.assert.status(404);
-                browser.assert.text('title', 'Error!!');
-                browser.assert.text('h1', 'It appears you messed up');
-            } else if (type === 'redirect-user') {
-                browser.assert.status(200);
-                browser.assert.text('title', 'Best redirect link ever');
-                browser.assert.text('h1', 'Welcome to your redirection file');
-            } else {
-                throw new Error('This is not checked in checkHTML()');
             }
-        }
+            return callback();
+        });
+    }
+
+    /**
+     * makeHeadRequest - makes head request and asserts expected response
+     * @param {(string|null)} auth - whether to use valid, invalid,
+     * or no authentication credentials
+     * @param {string} url - request url
+     * @param {number} expectedStatusCode - expected response code
+     * @param {object} expectedHeaders - expected headers in response with
+     * expected values (e.g., {x-amz-error-code: AccessDenied})
+     * @param {string | undefined} path - path for request or undefined if none
+     * @param {function} cb - callback to end test
+     * @return {undefined}
+     */
+    static makeHeadRequest(auth, url, expectedStatusCode, expectedHeaders, cb) {
+        _makeWebsiteRequest(auth, 'HEAD', url, (err, res) => {
+            // body should be empty
+            assert.deepStrictEqual(res.body, '');
+            assert.strictEqual(res.statusCode, expectedStatusCode);
+            const headers = Object.keys(expectedHeaders);
+            for (let i = 0; i < headers.length; i++) {
```

Comment:
this for loop is actually iterating through the header keys and comparing the value of header in result versus expected headers. I updated it to use `.forEach` since the for loop was a bit misleading.